### PR TITLE
ISPN-12628 All REST response descriptions should contain possible HTTP response codes

### PR DESCRIPTION
**Pull request by dmvolod**
All REST calls are reviews and documentation updated.
Also some redundant HTTP codes were changed in REST API description
@oraNod could you please review docs changes or/and add someone for review (I don't have an reviewer rights in this repo)

### CODE REVIEW

**Comment by oraNod** [comment]
> All REST calls are reviews and documentation updated.
> Also some redundant HTTP codes were changed in REST API description
> @oraNod could you please review docs changes or/and add someone for review (I don't have an reviewer rights in this repo)

@dmvolod I've added reviewers. This PR addresses you comment here, right? https://github.com/infinispan/infinispan-operator/issues/802#issuecomment-774066866 

Is it the case that the server image responds with `204` because that is set explicitly?

Also could you please squash? There are 14 commits here. Might make things a little cleaner. ;')

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:225]:
```suggestion
If a backup with the same name already exists, the service responds with `409 (Conflict)`.
If the `directory` parameter is not valid, the service returns `400 (Bad Request)`.
```

**Comment by dmvolod** [comment]
> I've added reviewers. This PR addresses you comment here, right? [infinispan/infinispan-operator#802 (comment)](https://github.com/infinispan/infinispan-operator/issues/802#issuecomment-774066866)
> 
Yes, @oraNod ,partially. I've reviewed all REST calls for possible response codes and found some missprints

> Is it the case that the server image responds with `204` because that is set explicitly?
> 
Yes, most of the codes setup explicitly in the REST server code

> Also could you please squash? There are 14 commits here. Might make things a little cleaner. ;')

Yes, sure, will squash. But will keep two commits with doc and code changes

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:128]:
```suggestion
If the entry is created successfully, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:384]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:483]:
```suggestion
* `sync` returns `204 (No Content)` only after the re-indexing operation is complete.
```

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:484]:
```suggestion
* `async` returns `204 (No Content)` immediately and the re-indexing operation continues running in the cluster. You can check the status with the link:#rest_v2_query_index_stats[Index Statistics] REST call.
```

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:495]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_clusters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_clusters.adoc:50]:
```suggestion
If the backup operation successfully completes, the service returns `202 (Accepted)`.
If a backup with the same name already exists, the service returns  `409 (Conflict)`.
If the `directory` parameter is not valid, the service returns `400 (Bad Request)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:87]:
```suggestion
`WEAK` counters never respond after operations and return `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:89]:
```suggestion
`STRONG` counters return `200 (OK)` and the current value after each operation.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:73]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:105]:
```suggestion
`WEAK` counters never respond after operations and return `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:107]:
```suggestion
`STRONG` counters return `200 (OK)` and the current value after each operation.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:125]:
```suggestion
`STRONG` counters return `200 (OK)` and the current value after each operation.
```

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:123]:
```suggestion
`WEAK` counters never respond after operations and return `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_logging.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_logging.adoc:47]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_logging.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_logging.adoc:60]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_protobuf.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_protobuf.adoc:25]:
```suggestion
If the operation successfully completes, the service returns `201 (Created)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_protobuf.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_protobuf.adoc:64]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_rolling.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_rolling.adoc:24]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

File: documentation/src/main/asciidoc/topics/ref_rest_servers.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_servers.adoc:48]:
```suggestion
{brandname} responds with `204 (No Content)` if the cache is successfully added to the ignore list or `404 (Not Found)` if the cache or cache manager are not found.
```

File: documentation/src/main/asciidoc/topics/ref_rest_servers.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_servers.adoc:66]:
```suggestion
{brandname} responds with `204 (No Content)` if the cache is successfully removed from ignore list or `404 (Not Found)` if the cache or cache manager are not found.
```

File: documentation/src/main/asciidoc/topics/ref_rest_xsite.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_xsite.adoc:194]:
```suggestion
If the operation successfully completes, the service returns `204 (No Content)`.
```

**Comment by dmvolod** [comment]
Thanks @oraNod for fix, commits squashed again

File: documentation/src/main/asciidoc/topics/ref_rest_counters.adoc
Comment (dmvolod) [inline documentation/src/main/asciidoc/topics/ref_rest_counters.adoc:45]: @oraNod please pay attention, that I've removed this call as this service not implemented as expected

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:263]: Why the change? AFAICT this will make create cache return 200 instead of 204?

**Review by gustavonalle — CHANGES_REQUESTED** [review]

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java
Comment (dmvolod) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:263]: No, it seems to be set always to `200` implicitly later with `responseBuilder.status(OK);` Please see L275, L285 and L307
it's just redundant call

Comment (gustavonalle) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:263]: ok, you are right, but it should be 204 in this case, too late to fix it now

**Comment by gustavonalle** [comment]
@oraNod waiting for your review before merging

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:131]:
```suggestion
`409 (Conflict)` and does not modify the value.
```

File: documentation/src/main/asciidoc/topics/ref_rest_caches.adoc
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_caches.adoc:192]:
```suggestion
return `409 (Conflict)` instead of modifying values. See
```

File: documentation/src/main/asciidoc/stories/assembly_rest_api.adoc
Comment (dmvolod) [reply documentation/src/main/asciidoc/stories/assembly_rest_api.adoc:6]: Hey, @oraNod 
It could be possible, but we need to say that some services can return `204` as success, but some `204`. Don't have an idea. We should describe a default service behavior.

Comment (oraNod) [reply documentation/src/main/asciidoc/stories/assembly_rest_api.adoc:6]: @dmvolod Cool. How about we just make this a note like:

[NOTE]
====
By default {brandname} REST API operations return `200 (OK)` when successful.

However, when some operations are processed successfully, they return an HTTP status code such as `204` or `202` instead of `200`.
====

Comment (dmvolod) [reply documentation/src/main/asciidoc/stories/assembly_rest_api.adoc:6]: Excellent, thanks @oraNod 
Will update soon

**Review by oraNod — APPROVED** [review]
LGTM

**Comment by oraNod** [comment]
@gustavonalle This PR seems fine to me. Should we go ahead and merge?

**Comment by oraNod** [comment]
@dmvolod Can you please squash?

**Comment by dmvolod** [comment]
@oraNod sure, done

**Review by gustavonalle — APPROVED** [review]

**Comment by oraNod** [comment]
thanks @dmvolod !

**Pull request merged**
